components/page-templates/B2B/AccountHierarchyTemplate/AccountHierarchyTemplate.spec.tsx

```diff
@@ -0,0 +1,153 @@
+  <div data-testid="account-hierarchy-form-mock">
+    <button data-testid="cancel-account-mock-button" onClick={onClose}>
+      Cancel
+    </button>
+    <button data-testid="save-account-mock-button" onClick={onClose}>
+      Save
+    </button>
+  </div>
+)
+jest.mock(
+  '@/components/b2b/AccountHierarchy/AccountHierarchyForm/AccountHierarchyForm',
+  () => () => AccountHierarchyMock({ onClose: onCloseMock })
+)
+
+const AccountHierarchyTreeMock = () => <div data-testid="account-hierarchy-tree-mock"></div>
+jest.mock(
+  '@/components/b2b/AccountHierarchy/AccountHierarchyTree/AccountHierarchyTree',
+  () => () => AccountHierarchyTreeMock()
+)
+
+jest.mock('@/components/dialogs', () => ({
+  __esModule: true,
+  AccountHierarchyDialog: (props: AccountHierarchyDialogProps) => {
+    const params = {
+      parentAccount: { id: 1023, companyOrOrganization: 'Parent Account' },
+      companyOrOrganization: 'ABCD',
+      taxId: '123234',
+      firstName: 'Karan',
+      lastName: 'Thappar',
+      emailAddress: 'karan@gmail.com',
+    }
+
+    return (
+      <div>
+        <h1>user-form-dialog</h1>
+        <button onClick={() => props.onSave(params)}>Confirm</button>
+      </div>
+    )
+  },
+}))
+
+jest.mock('next/router', () => ({
+  useRouter: jest.fn(),
+}))
+
+const setup = () => {
+  const user = userEvent.setup()
+  render(
+    <ModalContextProvider>
+      <Common />
+    </ModalContextProvider>,
+    {
+      wrapper: createQueryClientWrapper(),
+    }
+  )
+  return {
+    user,
+  }
+}
+
+describe('[component] - AccountHierarchyTemplate', () => {
+  it('should render component', async () => {
+    jest.mock('@/hooks', () => ({
+      useGetB2BUserQueries: jest.fn().mockReturnValue({
+        data: { id: 1023 },
+        isLoading: false,
+      }),
+    }))
+
+    render(<Common />, {
+      wrapper: createQueryClientWrapper(),
+    })
+
+    const heading = screen.getByText('account-hierarchy')
+    expect(heading).toBeVisible()
+
+    const addUserButton = screen.getByText('add-child-account')
+    expect(addUserButton).toBeVisible()
+
+    const accountHierarchyTreeMock = screen.getByTestId('account-hierarchy-tree-mock')
+    expect(accountHierarchyTreeMock).toBeVisible()
+  })
+
+  it('should open add child account form in dialog when add child account button clicked', async () => {
+    const { user } = setup()
+
+    const addChildAccountButton = screen.getByText('add-child-account')
+    user.click(addChildAccountButton)
+  })
+
+  it('should navigate to "/my-account" when account title is clicked', async () => {
+    window.matchMedia = createMatchMedia(1400)
+
+    const mockPush = jest.fn()
+    ;(useRouter as jest.Mock).mockReturnValue({
+      push: mockPush,
+    })
+
+    setup()
+
+    const accountTitleElement = screen.getByText(/my-account/i)
+    fireEvent.click(accountTitleElement)
+
+    expect(mockPush).toHaveBeenCalledWith('/my-account')
+  })
+})
```

Comment:
mock the AccountHierarchyFormDialog

Reply:
its already mocked with the name AccountHierarchyDialog
renamed it to AccountHierarchyFormDialog